external-libs/build/build-artifacts.sh

```diff
@@ -0,0 +1,552 @@
+BOOST_URL_BASE=${BOOST_URL_BASE:-https://sourceforge.net/projects/boost/files/boost}
+BOOST_URL=${BOOST_URL_BASE}/${BOOST_VERSION}/${BOOST_TAR}/download
+
+APK_ROOT=${APK_ROOT:-${BUILD_ROOT}/xmrwallet}
+XMRWALLET_GIT_URL=${XMRWALLET_GIT_URL:-https://github.com/m2049r/xmrwallet.git}
+
+declare -a BUILD_ARCHS=("arm" "arm64" "x86" "x86_64")
+declare -a BUILD_PACKAGES=("boost" "openssl" "monero")
+declare -a BUILD_ACTIONS=("download" "boost" "openssl" "monero" "apk" "checksum")
+# release or debug
+BUILD_TYPE=${BUILD_TYPE:-release}
+BUILD_CLEAN=${BUILD_CLEAN:-1}
+BUILD_CLEAN_ARTIFACT=${BUILD_CLEAN_ARTIFACT:-1}
+
+die() {
+  echo "$@"
+  exit 1
+}
+
+download() {
+  # Download files
+  # Fetch all remote files that are required for the build to avoid breaking
+  # the build when the remotes are temporary down.
+  # Check the sha256sum
+
+  if [ "${BUILD_CLEAN}" == "1" ]; then
+    rm -rf "${BUILD_ROOT}/distfiles"
+  fi
+  mkdir -p "${BUILD_ROOT}/distfiles"
+  pushd "${BUILD_ROOT}/distfiles"
+
+  test -f "${OPENSSL_TAR}" \
+    || wget "${OPENSSL_URL}" -O "${OPENSSL_TAR}" \
+      || die "cannot get ${OPENSSL_URL}"
+  checksum "${OPENSSL_TAR}" "${OPENSSL_TAR_SHA256SUM}"
+  test -f "${BOOST_TAR}" \
+    || wget "${BOOST_URL}" -O "${BOOST_TAR}" \
+      || die "cannot get ${BOOST_URL}"
+  checksum "${BOOST_TAR}" "${BOOST_TAR_SHA256SUM}"
+  test -d android-openssl \
+    && (cd android-openssl && git pull || die "android-openssl could not be updated") \
+    || git clone "${ANDROID_OPENSSL_GIT_URL}" \
+    || die "android-openssl could not be cloned"
+
+  # monero with the (master) submodules in the MONERO_GIT_BRANCH
+  test -d monero \
+    && (cd monero && git checkout master && git pull || die "monero could not be updated") \
+    || git clone --recursive "${MONERO_GIT_URL}" \
+    || die "monero could not be cloned"
+  pushd monero
+    if [ "${MONERO_GIT_BRANCH}" == "latest" ]; then
+      MONERO_GIT_BRANCH=$( \
+        git branch -a --sort=-committerdate \
+        | head -1 | tr -d '[:space:]')
+    fi
+    git checkout "${MONERO_GIT_BRANCH}"
+    git pull
+    git submodule init \
+      || die "monero submodules could not be initialized"
+    git submodule update \
+      || die "monero submodules could not be updated"
+  popd # monero
+
+  # xmrwallet for apk
+  test -d xmrwallet \
+    && (cd xmrwallet && git pull || die "xmrwallet could not be updated") \
+    || git clone "${XMRWALLET_GIT_URL}" \
+    || die "xmrwallet could not be cloned"
+  popd
+}
+
+checksum() {
+  # Checksum
+  # check the given filename has the given sha256sum, die when not the same
+  FILENAME="$1"
+  SHA256SUM="$2"
+  TMP_CHECKFILE="$(mktemp)"
+  echo -n "${SHA256SUM}  ${FILENAME}" \
+    > "${TMP_CHECKFILE}"
+  sha256sum -c "${TMP_CHECKFILE}" \
+    || die "checksum for ${FILENAME} not expected:" \
+      " is $(sha256sum "${FILENAME}") instead of ${SHA256SUM}"
+  # XX todo: should also be removed on die, irrelevant in container
```

Comment:
To avoid leaving temporary file, we can use pipe instead of file.
For instance, ```echo -n "${SHA256SUM}  ${FILENAME}" | sha256sum -c -```

Reply:
ok